tests/v1/e2e/test_spec_decode.py

```diff
@@ -144,3 +143,50 @@ def test_eagle_correctness(
         # Upon failure, inspect the outputs to check for inaccuracy.
         assert matches > int(0.7 * len(ref_outputs))
         del spec_llm
+
+
+def test_prefix_cache(
+    monkeypatch: pytest.MonkeyPatch,
+    test_prompts: list[list[dict[str, Any]]],
+    sampling_config: SamplingParams,
+    model_name: str,
+    eagle_model_name: str,
+):
+    with monkeypatch.context() as m:
+        m.setenv("VLLM_USE_V1", "1")
+
+        # Populate the cache with a few prompts
+        spec_llm = LLM(model=model_name,
+                       speculative_config={
+                           "method": "eagle",
+                           "model": eagle_model_name,
+                           "num_speculative_tokens": 3,
+                       },
+                       max_model_len=1024,
+                       disable_log_stats=False)
+
+        first_outputs = spec_llm.chat(test_prompts, sampling_config)
+
+        # Now, run the same prompts again to check if the cache is used
+        second_outputs = spec_llm.chat(test_prompts, sampling_config)
+
+        # Check:
+        # 1. The output is the almost same as the first run.
+        # 2. There is cache hit.
+        matches = 0
+        misses = 0
+        for first_output, second_output in zip(first_outputs, second_outputs):
+            if first_output.outputs[0].text == second_output.outputs[0].text:
+                matches += 1
+            else:
+                misses += 1
+                print(f"ref_output: {first_output.outputs[0].text}")
+                print(f"spec_output: {second_output.outputs[0].text}")
+
+        # Heuristic: expect at least 70% of the prompts to match exactly
+        # Upon failure, inspect the outputs to check for inaccuracy.
+        assert matches > int(0.9 * len(first_outputs))
```

Comment:
This test is too general. Ideally, we should get the cache hit metric/stats and spec decode stats. So that we can check
1. we reuse the kv cahe as expected.
2. the acceptance behavior is the same with/wo spec decode, so that we can make sure the input preparation logic of eagle is correct with prefix caching.